src/afrolabs/components/kafka.clj

```diff
@@ -466,6 +467,36 @@
         (#{:both :value} consumer-option)  (assoc ConsumerConfig/VALUE_DESERIALIZER_CLASS_CONFIG  "afrolabs.components.kafka.edn_serdes.Deserializer"
                                                   (:parse-inst-as-java-time -edn-serdes/config-keys) parse-inst-as-java-time)))))
 
+(defstrategy TransitSerializer
+  [& {producer-option :producer
+      consumer-option :consumer
+
+      :or   {producer-option         :none
+             consumer-option         :none}}]
+
+  (let [allowed-values #{:key :value :both :none}]
+    (when-not (or (allowed-values producer-option)
+                  (allowed-values consumer-option))
+      (throw (ex-info "TransitSerializer expects one of #{:key :value :both} for each of :producer or :consumer, eg (TransitSerializer :producer :both :consumer :key)"
+                      {::allowed-values  allowed-values
+                       ::consumer-option consumer-option
+                       ::producer-option producer-option}))))
+
+  (reify
+    IUpdateProducerConfigHook
+    (update-producer-cfg-hook
+        [_ cfg]
+      (cond-> cfg
+        (#{:both :key}   producer-option)  (assoc ProducerConfig/KEY_SERIALIZER_CLASS_CONFIG   "afrolabs.components.kafka.transit_serdes.Serializer")
+        (#{:both :value} producer-option)  (assoc ProducerConfig/VALUE_SERIALIZER_CLASS_CONFIG "afrolabs.components.kafka.transit_serdes.Serializer")))
+
+    IUpdateConsumerConfigHook
+    (update-consumer-cfg-hook
+        [_ cfg]
+      (cond-> cfg
+        (#{:both :key}   consumer-option)  (assoc ConsumerConfig/KEY_DESERIALIZER_CLASS_CONFIG    "afrolabs.components.kafka.transit_serdes.Deserializer")
+        (#{:both :value} consumer-option)  (assoc ConsumerConfig/VALUE_DESERIALIZER_CLASS_CONFIG  "afrolabs.components.kafka.transit_serdes.Deserializer")))))
```

Comment:
pulling in the new transit serdes class and defining a strategy for using it. We used this for the `location-replay` functionality in telemetry service.